.github/workflows/css_check.py

```diff
@@ -0,0 +1,147 @@
+            continue
+
+        for file in files:
+            file_path = os.path.abspath(os.path.join(root, file))
+
+            # Skip excluded files
+            if file_path in exclude_files:
+                continue
+
+            # Process TypeScript files
+            if file.endswith((".ts", ".tsx")) and "test" not in root:
+                try:
+                    with open(file_path, "r", encoding="utf-8") as f:
+                        content = f.read()
+                except (IOError, UnicodeDecodeError) as e:
+                    print(f"Error reading file {file_path}: {e}")
+                    continue
+
+                # Check for CSS imports with an improved regex pattern
+                css_imports = re.findall(
+                    r'import\s+.*?["\'](.*?\.css)["\'];', content
+                )
+                for css_file in css_imports:
+                    # Check if the CSS import matches the allowed patterns
+                    if any(css_file.endswith(pattern) for pattern in allowed_css_patterns):
+                        correct_css_imports.append(
+                            f"Correct CSS import ({css_file}) in {file_path}"
+                        )
+                    else:
+                        violations.append(
+                            f"Invalid CSS import ({css_file}) in {file_path}"
+                        )
+
+                # Check for embedded CSS
+                embedded_css = check_embedded_css(content)
+                if embedded_css:
+                    embedded_css_violations.append(
+                        f"Embedded CSS found in {file_path}: {', '.join(embedded_css)}"
+                    )
+
+    return violations, correct_css_imports, embedded_css_violations
+
+
+def main():
+    """Run the CSS check script."""
+    parser = argparse.ArgumentParser(
+        description="Check for CSS violations in TypeScript files."
+    )
+    parser.add_argument("--directory", required=True, help="Directory to check.")
+    parser.add_argument(
+        "--exclude_files",
+        nargs="*",
+        default=[],
+        help="Specific files to exclude from analysis.",
+    )
+    parser.add_argument(
+        "--exclude_directories",
+        nargs="*",
+        default=[],
+        help="Directories to exclude from analysis.",
+    )
+    parser.add_argument(
+        "--allowed_css_patterns",
+        nargs="*",
+        default=["app.module.css"],
+        help="Allowed CSS file patterns.",
+    )
+    args = parser.parse_args()
+
+    violations, correct_css_imports, embedded_css_violations = check_files(
+        directory=args.directory,
+        exclude_files=args.exclude_files,
+        exclude_directories=args.exclude_directories,
+        allowed_css_patterns=args.allowed_css_patterns,
+    )
+
+    if violations:
+        print("\nCSS Import Violations:")
+        print("\n".join(violations))
+
+    if embedded_css_violations:
+        print("\nEmbedded CSS Violations:")
```

Comment:
1. Use a single string and at the end give an explanation of what the contributor must do. Use the python """ string representation to achieve this.
    1. We need contributors to understand the next steps and fix things without having to ask why
    2. It reduces the workload of the reviewers
3. Give the color code and line number in the message. Make it easier for the end user to rectify the situation